dev/src/index.ts

```diff
@@ -1121,77 +1174,69 @@ follow these steps, YOUR APP MAY BREAK.`);
+   * retried.
+   * @returns A Promise with the resulting read/write stream.
    */
-  readWriteStream(methodName, request, requestTag, allowRetries) {
-    let self = this;
-    let attempts = allowRetries ? MAX_REQUEST_RETRIES : 1;
+  readWriteStream(
+      methodName: string, request: {}, requestTag: string,
+      allowRetries: boolean): Promise<NodeJS.ReadWriteStream> {
+    const self = this;
+    const attempts = allowRetries ? MAX_REQUEST_RETRIES : 1;
 
     return this._runRequest(gapicClient => {
       const decorated = this._decorateRequest(request);
       return this._retry(attempts, requestTag, () => {
         return Promise.resolve().then(() => {
           logger('Firestore.readWriteStream', requestTag, 'Opening stream');
-          let requestStream = gapicClient[methodName](decorated.gax);
+          const requestStream = gapicClient[methodName](decorated.gax);
 
           // The transform stream to assign the project ID.
-          let transform = through2.obj(function(chunk, encoding, callback) {
-            let decoratedChunk = extend(true, {}, chunk);
+          const transform = through2.obj((chunk, encoding, callback) => {
+            const decoratedChunk = extend(true, {}, chunk);
             replaceProjectIdToken(
-                decoratedChunk, self._referencePath.projectId);
+                decoratedChunk, self._referencePath!.projectId);
             logger(
                 'Firestore.readWriteStream', requestTag,
                 'Streaming request: %j', decoratedChunk);
             requestStream.write(decoratedChunk, encoding, callback);
           });
 
-          let logStream = through2.obj(function(chunk, enc, callback) {
+          const logStream = through2.obj(function(
+              this: AnyDuringMigration, chunk, enc, callback) {
             logger(
                 'Firestore.readWriteStream', requestTag,
                 'Received response: %j', chunk);
             this.push(chunk);
             callback();
           });
 
-          let resultStream = bun([transform, requestStream, logStream]);
+          const resultStream = bun([transform, requestStream, logStream]);
           return this._initializeStream(resultStream, requestTag, request);
         });
       });
     });
   }
 }
 
-
 /**
  * Validates a JavaScript value for usage as a Firestore value.
  *
  * @private
- * @param {Object} val JavaScript value to validate.
- * @param {string} options.allowDeletes At what level field deletes are
- * supported (acceptable values are 'none', 'root' or 'all').
- * @param {boolean} options.allowServerTimestamps Whether server timestamps
- * are supported.
- * @param {number=} depth The current depth of the traversal.
- * @param {number=} inArray Whether we are inside an array.
- * @returns {boolean} 'true' when the object is valid.
- * @throws {Error} when the object is invalid.
+ * @param val JavaScript value to validate.
+ * @param  options Validation options
```

Comment:
nit: Maybe grep the code for "@param  " because I know there's at least a couple of these.

Reply:
Fixed 8 of them.